### PR TITLE
DR.WB SSR LIVE RELOADING

### PR DESCRIPTION
**Pull request by deadxears**
GOING INTO ALPHA

### CODE REVIEW

**Review by dwaynerichards — APPROVED** [review]

**Review by dwaynerichards — APPROVED** [review]
reviewed.

**Review by dwaynerichards — APPROVED** [review]
reviewed

**Review by dwaynerichards — APPROVED** [review]

**Review by HeroesOfOnesAndZeros — APPROVED** [review]
reviewed

File: core/cli/templates.ts [hunk 1]
```diff
@@ -145,3 +146,45 @@ export const vnoConfig = (options: CreateInputs) => {
     2,
   );
 };
+
+export const ssrTemplate =
+  `import { opine, serveStatic } from "https://deno.land/x/opine@1.2.0/mod.ts";
+import  vueServerRenderer from 'https://deno.land/x/vue_server_renderer@/mod.js';
+
+import App from './vno-ssr/build.js';
+import { join, dirname} from "https://deno.land/std@0.63.0/path/mod.ts";
```
Comment (buttercubz) [inline core/cli/templates.ts:155]: I could update to [`std@0.93.0`](https://deno.land/std@0.93.0) which already has many bug solved

File: core/cli/commands.ts [hunk 2]
```diff
@@ -1,16 +1,25 @@
...
+import info from './info.ts';
+import * as fn from './fns.ts';
+import { fs, path } from '../utils/deps.ts';
+import { createApplication } from './create.ts';
+import { runDevServer } from './dev.ts';
+import { quietArg } from './fns.ts';
+import { cmnd, serverTs, vnoconfig } from './constants.ts';
+import { Config } from '../dts/factory.d.ts';
+import { ssrTemplate } from '../cli/templates.ts';
+import { existsSync } from 'https://deno.land/std/fs/mod.ts';
```
Comment (buttercubz) [inline core/cli/commands.ts:12]: try not to use unversioned url to avoid breaking changes

File: core/utils/livereload.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+// Connection opened
+socket.addEventListener("open", function (event) {
+  socket.send("[LiveReload client connected]");
+});
+
+// Listen for messages
+socket.addEventListener("message", function (event) {
+  socket.send("[LiveReload reloading...]");
+  if (event.data === 'window.location.reload();') {
+    eval(event.data);
```
Comment (buttercubz) [inline core/utils/livereload.ts:16]: please don't use `eval()` it's a very risk

File: core/cli/dev.ts [hunk 4]
```diff
@@ -1,9 +1,33 @@
...
 export const server: Application = new Application();
 
 export const runDevServer = async function (port: number, hostname: string) {
+  const wss = new WebSocketServer(8080);
+  wss.on("connection", function (ws: WebSocketClient) {
+    ws.send('[LiveReload is watching...');
+    // create event listeneer that listens for "buildDone" event
+    const reloadWindow = () => {
+      console.log("[back to you Client!]");
+      ws.send('window.location.reload();');
```
Comment (buttercubz) [inline core/cli/dev.ts:19]: https://github.com/open-source-labs/vno/pull/106/files#r614435145

**Review by buttercubz — CHANGES_REQUESTED** [review]
For now looks promising

**Comment by unknownbreaker** [comment]
Thanks for giving a once over @buttercubz, and great points regarding `eval` and versioning. We'll revisit this.

**Review by buttercubz — APPROVED** [review]

**Comment by HeroesOfOnesAndZeros** [comment]
fixed errors, going to resubmit pull request from oslabs beta

**Pull request closed without merging**